### DISCUSSION TITLE
gitlab integration

### DISCUSSION THREAD

**Discussion by ChristopherPisz** [friendlyanon/cmake-init#64] [Q&A]
Spent the day reading over gitlab docs. Just trying to get the build and static analysis tools working. Will worry about making my project available via conan after I can actually get an auto mated build.

It seems cmake-init generated a yml file for github, but I think that file differs from the yml gitlab wants. I also think gitlab wants to run everything in a docker container.

Not sure where to start with this, but playing around with it.
 
Does there already exist a docker container that has everything needed (static analysis tools, python, and conan) or do I need to learn docker this week too? I'm gonna need a raise if I am gonna learn all this devops stuff!

**Comment by friendlyanon** [comment]
I don't know whether GitLab provides free CI with Linux, Windows and MacOS runners. When I looked, I couldn't find a clear answer. I could add a GitLab CI workflow as well, but that's going to happen only if those 3 platforms can be tested.

GitHub is extremely clear on this and the project is hosted on this platform anyway, so it was a no-brainer to provide.

**Reply by ChristopherPisz** [reply]
From what I can tell, it absolutely provides a free version for those that want to manage their own instance. https://about.gitlab.com/install/#ubuntu
